app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -985,7 +985,9 @@ protected function sendEmailConfirmation(CustomerInterface $customer, $redirectU
                 $templateType = self::NEW_ACCOUNT_EMAIL_REGISTERED_NO_PASSWORD;
             }
             $this->getEmailNotification()->newAccount($customer, $templateType, $redirectUrl, $customer->getStoreId());
-            $customer->setConfirmation(null);
+            if ($templateType != self::NEW_ACCOUNT_EMAIL_CONFIRMATION) {
```

Comment:
Could you replace it to strict comparison?
```suggestion
            if ($templateType !== self::NEW_ACCOUNT_EMAIL_CONFIRMATION) {
```